nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/util/orc/OrcUtils.java

```diff
@@ -0,0 +1,443 @@
+                byte[] buf = new byte[size];
+                byteBuffer.get(buf, 0, size);
+                ((BytesColumnVector) col).setVal(rowNumber, buf);
+            }
+        } else if (Schema.Type.DOUBLE.equals(fieldType)) {
+            if (o == null) {
+                col.isNull[rowNumber] = true;
+            } else {
+                ((DoubleColumnVector) col).vector[rowNumber] = (double) o;
+            }
+        } else if (Schema.Type.FLOAT.equals(fieldType)) {
+            if (o == null) {
+                col.isNull[rowNumber] = true;
+            } else {
+                ((DoubleColumnVector) col).vector[rowNumber] = (float) o;
+            }
+        } else if (Schema.Type.STRING.equals(fieldType) || Schema.Type.ENUM.equals(fieldType)) {
+            if (o == null) {
+                col.isNull[rowNumber] = true;
+            } else {
+                ((BytesColumnVector) col).setVal(rowNumber, o.toString().getBytes());
+            }
+        } else if (Schema.Type.UNION.equals(fieldType)) {
+            // If the union only has one non-null type in it, it was flattened in the ORC schema
+            if (col instanceof UnionColumnVector) {
+                UnionColumnVector union = ((UnionColumnVector) col);
+                Schema.Type avroType = OrcUtils.getAvroSchemaTypeOfObject(o);
+                // Find the index in the union with the matching Avro type
+                int unionIndex = -1;
+                List<Schema> types = fieldSchema.getTypes();
+                final int numFields = types.size();
+                for (int i = 0; i < numFields && unionIndex == -1; i++) {
+                    if (avroType.equals(types.get(i).getType())) {
+                        unionIndex = i;
+                    }
+                }
+                if (unionIndex == -1) {
+                    throw new IllegalArgumentException("Object type " + avroType.getName() + " not found in union '" + fieldSchema.getName() + "'");
+                }
+
+                // Need nested vector offsets
+                MutableInt unionVectorOffset = new MutableInt(0);
+                putToRowBatch(union.fields[unionIndex], unionVectorOffset, rowNumber, fieldSchema.getTypes().get(unionIndex), o);
+            } else {
+                // Find and use the non-null type from the union
+                List<Schema> types = fieldSchema.getTypes();
+                Schema effectiveType = null;
+                for (Schema type : types) {
+                    if (!Schema.Type.NULL.equals(type.getType())) {
+                        effectiveType = type;
+                        break;
+                    }
+                }
+                putToRowBatch(col, vectorOffset, rowNumber, effectiveType, o);
+            }
+
+        } else if (Schema.Type.ARRAY.equals(fieldType)) {
+            Schema arrayType = fieldSchema.getElementType();
+            ListColumnVector array = ((ListColumnVector) col);
+            if (o instanceof int[]) {
```

Comment:
The logic for `int[]` and `long[]` appear to be the same and we are using the LongColumnVector in both cases. Does it make sense to collapse these two clauses into just a single conditional: `if (o instanceof int[] || o instanceof long[]) {`